### ISSUE TITLE
src/mux_h2.c: unused value suspected by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#2752]
### Tool Name and Version

coverity

### Code Report

```plain
/src/mux_h2.c: 6060 in h2_frt_transfer_data()
6054                    h2s->body_len -= sent;
6055                    htx->extra = h2s->body_len;
6056            }
6057     
6058            if (sent < flen) {
6059                    if (!sent)
>>>     CID 1563741:    (UNUSED_VALUE)
>>>     Assigning value "1" to "full" here, but that stored value is overwritten before it can be used.
6060                            full = 1;
6061                    if (h2s_get_rxbuf(h2s))
6062                            goto next_buffer;
6063     
6064                    h2c->flags |= H2_CF_DEM_SFULL;
6065                    TRACE_STATE("h2s rxbuf is full", H2_EV_RX_FRAME|H2_EV_RX_DATA|H2_EV_H2S_BLK, h2c->conn, h2s);
/src/mux_h2.c: 6029 in h2_frt_transfer_data()
6023                    if (!flen)
6024                            goto fail;
6025            }
6026     
6027            block = htx_free_data_space(htx);
6028            if (!block) {
>>>     CID 1563741:    (UNUSED_VALUE)
>>>     Assigning value "1" to "full" here, but that stored value is overwritten before it can be used.
6029                    full = 1;
6030                    if (h2s_get_rxbuf(h2s))
6031                            goto next_buffer;
6032     
6033                    h2c->flags |= H2_CF_DEM_SFULL;
6034                    TRACE_STATE("h2s rxbuf is full", H2_EV_RX_FRAME|H2_EV_RX_DATA|H2_EV_H2S_BLK, h2c->conn, h2s);
```

### Output of `haproxy -vv`

```plain
no
```

**Comment by wtarreau** [comment]
Good catch, it was indeed dropped between 11th and 12th iteration of the rxwin dev, but the variable was still kept. I've removed it now. Will push soon (other fixes in my branch before merging).

Thank you!

**Comment by capflam** [comment]
It was fixed by db76949cff312619a1aa76c10622362eb5b9997b
